File: java/src/main/TA_Entity.java
package main;

import com.tictactec.ta.lib.MInteger;

public class TA_Entity {
	
	private MInteger begin;
	private MInteger length;
	private double[] prices;
	
	public TA_Entity(MInteger begin, MInteger length, double[] prices)
	{
		this.begin = begin;
		this.length = length;
		this.prices = prices;
	}
	
	public int getBegin() { return begin.value; }
	public int getLength() { return length.value; }
	public double[] getPrices() { return prices; }
	
	public double get(int id) throws TA_Exception { 
		if(id >= begin.value && id <= (begin.value+length.value))
			return prices[id]; 
		else
			throw new TA_Exception("Invalid id : " + id);
	}
	
}

File: src/TA_Entity.cpp
#include "../include/TA_Entity.h"

using namespace std;

TA_Entity::TA_Entity(TA_Real begin, TA_Real nbElements, TA_Real* out)
{
    _begin = begin;
    _nbElements = nbElements;

    for(int i= 0; i< _nbElements; i++) {
        int k = _begin + i;
        //cout << out[k] << endl;
        _out[k] = out[k];
       // cout << _out[k] << endl;
    }

}

TA_Real TA_Entity::getBegin() { return _begin; }
TA_Real TA_Entity::getNbElements() { return _nbElements; }
TA_Real* TA_Entity::getPrices() { return _out; }
TA_Real TA_Entity::get(int id) { return _out[id]; }


TA_Entity::~TA_Entity() { delete(_out); }
